lib/main.dart

```diff
@@ -0,0 +1,210 @@
+import 'package:english_words/english_words.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+void main() {
+  runApp(const MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return ChangeNotifierProvider(
+      create: (context) => MyAppState(),
+      child: MaterialApp(
+        title: 'Namer App',
+        theme: ThemeData(
+          useMaterial3: true,
+          colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepOrange),
+        ),
+        home: const MyHomePage(),
+      ),
+    );
+  }
+}
+
+class MyAppState extends ChangeNotifier {
+  var current = WordPair.random();
+
+  void getNext() {
+    current = WordPair.random();
+    notifyListeners();
+  }
+
+  var favorites = <WordPair>[];
+
+  void toggleFavorite() {
+    if (favorites.contains(current)) {
+      favorites.remove(current);
+    } else {
+      favorites.add(current);
+    }
+    notifyListeners();
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  const MyHomePage();
+
+  @override
+  State<MyHomePage> createState() => _MyHomePageState();
+}
+
+class FavoritesPage extends StatelessWidget {
+  const FavoritesPage();
+
+  @override
+  Widget build(BuildContext context) {
+    final appState = context.watch<MyAppState>();
+
+    if (appState.favorites.isEmpty) {
+      return const Center(
+        child: Text('No favorites yet.'),
+      );
+    }
+
+    return ListView(
+      children: [
+        Padding(
+          padding: const EdgeInsets.all(20),
+          child: Text('You have '
+              '${appState.favorites.length} favorites:'),
+        ),
+        for (var pair in appState.favorites)
+          ListTile(
+            leading: const Icon(Icons.favorite),
+            title: Text(pair.asLowerCase),
+          ),
+      ],
+    );
+  }
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  var selectedIndex = 0;
+
+  Widget _getPageByIndex(int index) {
+    Widget page;
+    switch (index) {
+      case 0:
+        page = GeneratorPage();
+        break;
+      case 1:
+        page = const FavoritesPage();
+        break;
+      default:
+        throw UnimplementedError('no widget for $index');
+    }
+    return page;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final Widget page = _getPageByIndex(selectedIndex);
```

Comment:
```suggestion
    final page = _getPageByIndex(selectedIndex);
```

выведение типов в дарте хорошо работает, поэтому для локальных можно пропускать в таком случае

Reply:
Хорошо, а в каких тогда лучше указывать?

Reply:
да по идее для всех случаев кроме "локальной переменной".

Есть правило в линтере, обычно оно подсвечивает что да как)
https://dart-lang.github.io/linter/lints/omit_local_variable_types.html

Просто чуть меньше кода писать и менять приходится по необходимости )

Reply:
Понял! Спасибо!)